File: clj-from-java/src/main/clojure/capclug/demo/cljfromjava/seven.clj
(ns capclug.demo.cljfromjava.seven
  (:gen-class
    :methods [[foo ["[Ljava.lang.String;" ] String]]))
; tried #^(into-array String nil)
; tried #^"[Ljava.lang.String;"
; tried Array
; tried (meta #^#=(array-of String) [])
; tried #^#=(array-of String)
; tried #^String[]

(defn -foo [$^"[Ljava.lang.String;" args]
  (println "at least I got here! " args)
  "return me")

File: java-call-clj/src/main/java/capclug/demo/callclj/Seven.java
/*
 * $Id$
 * Copyright (C) 2010 ARIN
 */
package capclug.demo.callclj;

import capclug.demo.cljfromjava.seven;

/**
 * @version $Rev$
 */
public class Seven
{
    public static void main( String[] args )
    {
        seven seven = new seven();
        seven.foo( new String[] { "la", "ti", "da"} );
    }
}
